invenio_oaiharvester/api.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+from sickle import Sickle
+from invenio_oaiharvester.errors import (NameOrUrlMissing, WrongDateCombination,
+                                         NAME_OR_URL_ERROR_MSG, WRONG_DATES_ERROR_MSG)
+from invenio_oaiharvester.models import OaiHARVEST
+
+
+def list_records(metadata_prefix, from_date=None, until_date=None, url=None, name=None, setSpec=None):
+    """
+    Harvest records from an OAI repo, based on datestamp and/or set parameters.
+    :param metadata_prefix: The prefix for the metadata return (e.g. 'arXiv') (required).
+    :param from_date: The lower bound date for the harvesting (optional).
+    :param until_date: The upper bound date for the harvesting (optional).
+    :param url: The The url to be used to create the endpoint.
+    :param name: The name of the OaiHARVEST object that we want to use to create the endpoint.
+    :param setSpec: The 'set' criteria for the harvesting (optional).
+    :return: A list of harvested records.
+    """
+    request = make_endpoint_request(url, name)
+
+    if from_date is None:   # then we are using object.lastrun
+        obj = get_oaiharvest_obj(name)
+        from_date = obj.lastrun
+        update_lastrun(obj)
+
+    dates = {'from': from_date}\
+        if until_date is None \
+        else {'from': from_date, 'until': until_date}
+
+    # Sanity check
+    if from_date > until_date:
+        raise WrongDateCombination(WRONG_DATES_ERROR_MSG)
+
+    return list(request.ListRecords(metadataPrefix=metadata_prefix, set=setSpec, **dates))
+
+
+def get_records(metadata_prefix, identifiers, url=None, name=None):
```

Comment:
Same as https://github.com/inspirehep/invenio-oaiharvester/pull/1/files#r34669907 , the `metadata_prefix` can be an optional parameter if derived from `name`.